src/agent/datapath_reducer.rs

```diff
@@ -67,6 +71,18 @@ impl Reducible for DatapathReducer {
                         current_stage: stage,
                         ..self.mips.clone()
                     },
+                    MipsStateUpdate::UpdateSpeed(speed) => MipsCoreState {
```

Comment:
While implementing the syscalls I made a different variant of the state updates enum for system-related calls, that way we don't have to reimplement them for every single architecture.

https://github.com/SWIM-ucf/SWIM/blob/bc98bade8ee402be78498c13e96976db731958e5/src/agent/messages.rs

If you're good with it we can add UpdateSpeed, UpdateExecuting, and UpdateInitialized as variants of that enum.

Reply:
That sounds good. I've merged my branch into yours to make the changes and removed `UpdateSpeed` since the front-end doesn't actually need it.